exasol/analytics/schema/statements.py

```diff
@@ -0,0 +1,128 @@
+from __future__ import annotations
+
+from typing import Any
+
+from exasol.analytics.schema.column_name import ColumnName
+from exasol.analytics.schema.values import quote_value
+
+
+class UnknownColumnError(Exception):
+    """
+    In case of adding a value for a column not contained in the initial
+    list of available columns.
+    """
+
+
+class InsertStatement:
+    """
+    Enables adding columns and values for creating an SQL INSERT statement.
+
+    All available columns need to be specified up front when creating an
+    instance of this class.
+
+    Trying to add a value for a column missing in the initial list will raise
+    an exception.
+
+    Let's look at an example for an INSERT statement:
+
+        INSERT INTO "T" ("C1", "C2", "C3", "C4")
+        SELECT SYSTIMESTAMP(), 'Phase', SQ.B, SQ.A
+        FROM VALUES (1)
+        CROSS JOIN (SELECT "A", "B" FROM S.T2) as SQ
+
+    Here we see 3 columns "C1", "C2", and "R" being inserted into table "T"
+    while the values after `SELECT` have 3 different categories:
+
+    * SYSTIMESTAMP() is an SQL scalar function that must not be quoted.
+    * "Phase" is a string constant, that must be enclosed in single-quotes.
+    * SQ.B and SQ.A are references to columns "B" and "A" in a subquery with the alias "SQ".
+
+    Alias "SQ" is assigned to the subquery at the very end of the INSERT
+    statement. Please note that the alias also needs to match the name of the
```

Comment:
This statement is outdated I guess.

Reply:
Thanks fixed, with some additional things, that changed during review but I did forget to change in the docstring.